sdks/python/apache_beam/typehints/typehints_test.py

```diff
@@ -1610,5 +1610,27 @@ def test_pipe_operator_as_union(self):
           native_type_compatibility.convert_to_beam_type(type_b))
 
 
+class TestBuiltinsTyping(unittest.TestCase):
+
+  def _assert_equal_convert_to_beam_type(self, type_a, type_b):
+      beam_type_a = native_type_compatibility.convert_to_beam_type(type_a)
```

Comment:
These test would fail on Python 3.8 or lower since built-in type modules are supported from Python 3.9.

```
def test_convert_to_beam_type(self):
>       self._assert_equal_convert_to_beam_type(dict[str, int], typing.Dict[str, int])
E       TypeError: 'type' object is not subscriptable
```

Reply:
I guess the changes made are compatible with Python 3.9 or above. @jrmccluskey is working on similar(same may be) issue.

Reply:
The proper way to handle this is to enable the unit tests and the behavior only on Python 3.9 and higher. #24986 does this for the added unit test cases, so this should be okay to do here

Reply:
@DavidKatz-il Could you rebase on to HEAD and then run the tests?